var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -492,6 +495,18 @@ def bjam_python_line(self, spec):
             spec["python"].libs[0],
         )
 
+    def bjam_python_line_win(self, spec):
+        # avoid "ambiguous key" error
+        if spec.satisfies("@:1.58"):
+            return ""
```

Comment:
Can you explain what this is? Do you think there is another mechanism that should be used to specify Python before `1.59`? Or perhaps we can just say that `require(@1.59:, when="platform=windows")`?

Reply:
This is [in](https://github.com/spack/spack/blob/a7444873b9fa9a5837ba0a0853845b6691859051/var/spack/repos/builtin/packages/boost/package.py#L486) the regular `bjam_python_line` method that I copied , so I don't know why it was added originally. I can remove it if it is not useful.

Reply:
Ok I put them back into the same singular `bjam_python_line` method, but the actual paths are different on windows, so I just made the return conditional on OS instead.